### PR TITLE
feat: add NOT options for url targeting

### PR DESCRIPTION
**Pull request by shawnlikescode**
## Problem

Part of completing [Post/posthog/#22281](https://github.com/PostHog/posthog/issues/22281). 

related to [THIS PR](https://github.com/PostHog/posthog-js/pull/1219)

## Changes

image

### CODE REVIEW

**Review by neilkakkar — COMMENTED** [review]
just one comment, thanks for the contribution @shawnlikescode !

File: frontend/src/types.ts [hunk 1]
```diff
@@ -2546,8 +2546,11 @@ export interface Survey {
 
 export enum SurveyUrlMatchType {
     Exact = 'exact',
+    NotExact = 'not_exact',
```
[the remark this answers is not in the document]
Comment (shawnlikescode) [reply frontend/src/types.ts:2549]: For reuse are you looking for something like this?

```
export enum SurveyUrlMatchType {
    Exact = PropertyOperator.Exact,
    IsNot = PropertyOperator.IsNot,
    Contains = PropertyOperator.IContains,
    NotIContains = PropertyOperator.NotIContains,
    Regex = PropertyOperator.Regex,
    NotRegex = PropertyOperator.NotRegex,
}
```

```
export const SurveyUrlMatchTypeLabels = {
    [SurveyUrlMatchType.Exact]: allOperatorsMapping[SurveyUrlMatchType.Exact],
    [SurveyUrlMatchType.IsNot]: allOperatorsMapping[SurveyUrlMatchType.IsNot],
    [SurveyUrlMatchType.Contains]: allOperatorsMapping[SurveyUrlMatchType.Contains],
    [SurveyUrlMatchType.NotIContains]: allOperatorsMapping[SurveyUrlMatchType.NotIContains],
    [SurveyUrlMatchType.Regex]: allOperatorsMapping[SurveyUrlMatchType.Regex],
    [SurveyUrlMatchType.NotRegex]: allOperatorsMapping[SurveyUrlMatchType.NotRegex],
};
```

Also I had `NotContains` I see in that PropertyOperator that `NotIContains` is probably what we want to go with

[the remark this answers is not in the document]
Comment (neilkakkar) [reply frontend/src/types.ts:2549]: Indeed, exactly this :) - and good catch with the icontains!

[the remark this answers is not in the document]
Comment (shawnlikescode) [reply frontend/src/types.ts]: There we go!

[the remark this answers is not in the document]
Comment (shawnlikescode) [reply frontend/src/types.ts]: @neilkakkar Anything else you think I should complete here?

[the remark this answers is not in the document]
Comment (neilkakkar) [reply frontend/src/types.ts]: Nope looks good! Can you just merge master into your PR, I'll merge the posthog-js change and then merge this, thanks!

[the remark this answers is not in the document]
Comment (shawnlikescode) [reply frontend/src/types.ts]: Awesome.  Merged.

[the remark this answers is not in the document]
Comment (shawnlikescode) [reply frontend/src/types.ts]: @neilkakkar I just reset and tried the merge again. I think managed to not scuff it up this time.

[the remark this answers is not in the document]
Comment (neilkakkar) [reply frontend/src/types.ts]: just one remaining change it seems - I'll just revert that

File: frontend/src/scenes/pipeline/hogfunctions/pipelineHogFunctionConfigurationLogic.tsx [hunk 2]
```diff
@@ -253,7 +253,7 @@ export const pipelineHogFunctionConfigurationLogic = kea<pipelineHogFunctionConf
                 enabled: false,
             })
         },
-        loadHogFunctionSuccess: ({ hogFunction }) => actions.resetForm(hogFunction!),
+        loadHogFunctionSuccess: ({ hogFunction }) => actions.resetForm(hogFunction),
```
Comment (neilkakkar) [inline frontend/src/scenes/pipeline/hogfunctions/pipelineHogFunctionConfigurationLogic.tsx:256]:
```suggestion
        loadHogFunctionSuccess: ({ hogFunction }) => actions.resetForm(hogFunction!),
```

**Review by neilkakkar — APPROVED** [review]

**Pull request merged**